src/main/java/frc/robot/subsystems/climberSubsystem/ClimberSubsystem.java

```diff
@@ -132,10 +121,59 @@ public CurrentVelocity getVelocity() {
         leftArmMotor.getEncoder().getVelocity(), rightArmMotor.getEncoder().getVelocity());
   }
 
-  public AppliedCurrent getCurrent() {
-    return new AppliedCurrent(leftArmMotor.getOutputCurrent(), rightArmMotor.getOutputCurrent());
+  protected ClimberState getCurrentState() {
+    return state;
+  }
+
+  protected void setState(ClimberState pos) {
+    this.state = pos;
+    updateMotors();
   }
 
+  protected void advanceState() {
+    switch (state) {
```

Comment:
[Java Switch statement documentation]https://www.w3schools.com/java/java_switch.asp).

A `switch` statement with no `break` statements executes each of the items in succession.

Also: you should create a `default` option in case this gets called with `state == CLIP`. It can be a noop or just a log statement, but it should exist.